CPPLINT.cfg

```diff
@@ -1,12 +1,15 @@
 set noparent
-filter=-build/include_dirname,-build/c++11
+filter=-build/include_subdir
+filter=-build/c++11
 filter=-build/include_what_you_use
 filter=-legal/copyright
 filter=-runtime/int
 filter=-runtime/references
-filter=-readability/completely_reasonable_casts
 filter=-readability/todo
 filter=-readability/function
+filter=-build/header_guard
```

Comment:
won't these filters ignore things we care about?

Reply:
I disabled them because they cause a deluge of complaints. The customized pylint uses a different style from the recommended one. I readded some of the customizations, but they cause issues. So the next push will definitely break.

Actually, I am confused about the existence of [some .hpp files](https://github.com/p4lang/p4c/tree/master/frontends/parsers/p4) in the compiler. They are the only ones in the code base. Should they be converted to h files?